### DISCUSSION TITLE
[AIR 51.0.0.2] `EncryptedLocalStore` extra files?

### DISCUSSION THREAD

**Discussion by itlancer** [airsdk/Adobe-Runtime-Support#3095] [Q&A]
I noticed that in AIR 51 was made some changes around ELS:
"AIR-6063: Updated OpenSSL-based ELS key storage" https://airsdk.dev/news/2024/02/26/air-release

Before AIR 51 `EncryptedLocalStore` files stored at location (Windows example):
`c:\Users\username\AppData\Roaming\my.app.id\ELS\`
With AIR 51 `EncryptedLocalStore` files now stored at file:
`c:\Users\username\AppData\Roaming\Adobe\AIR\ELS\my.app.id.dat`

But using `EncryptedLocalStore` AIR 51 it still additionally save data to old directory `c:\Users\username\AppData\Roaming\my.app.id\ELS\`

Code example:
```
var bytes:ByteArray = EncryptedLocalStore.getItem("test");
trace(bytes);
bytes = new ByteArray();
bytes.writeUTFBytes("test");
EncryptedLocalStore.setItem("test", bytes);
```

My questions:
1) Why ELS still save files to "old" directory too? Maybe it just a bug.
Cause if I delete "old" directory when using `EncryptedLocalStore` with AIR 51 - nothing changes. All still working (using new .dat file).
2) Does new behavior related to all platforms? I mean storing at new .dat file.

**Comment by itlancer** [comment]
@ajwfrost 
Also I wonder where such .dat files stored for Android and iOS/tvOS platforms if it applicable?

**Reply by ajwfrost** [reply]
Android: app-specific persistent storage:
https://developer.android.com/reference/android/content/pm/ApplicationInfo#dataDir

xOS: user-specific persistent storage:
https://developer.apple.com/documentation/foundation/nssearchpathdirectory/nsapplicationsupportdirectory

To answer the earlier questions:
- the current code still uses the old mechanisms as a fallback for when it can't find information in the 'new' file. So it should work fine without that, and it shouldn't actually write to the old file at all, but for migration purposes, the old location will still be checked.
- yes this is all now the same across all platforms, the differences being in the key storage for the encryption that's used for the store file..

Hope that helps!

thanks
